tretools/codelists/codelist.py

```diff
@@ -5,22 +5,33 @@
 import os
 from typing import Optional, List, Dict
 import re
-
+import polars as pl
 
 from tretools.codelists.codelist_types import CodelistType
 from tretools.codelists.errors import InvalidSNOMEDCodeError, RepeatedCodeError, InvalidDataShapeError, InvalidICD10CodeError, InvalidOPCSCodesError
 
 class Codelist:
-    def __init__(self, path: str, codelist_type: CodelistType, code_column: str = "code", term_column: str = "term", add_x_codes: bool = False) -> None:
+    def __init__(self, path: str,
+                 codelist_type: CodelistType,
+                 code_column: str = "code",
+                 term_column: str = "term",
+                 snomed_to_icd10_path: str = '',
+                 add_x_codes: bool = False,
+                 snomed_to_icd10: bool = False,
+                 ICD10_3Digit: bool = False) -> None:
         self.codelist_type = codelist_type
         self.code_column = code_column
         self.term_column = term_column
         self.codes = set()
-
+        
         if add_x_codes and self.codelist_type == "ICD10":
             self.data = self._load_codelist(path, add_x_codes)
         else:
             self.data = self._load_codelist(path)
+        if snomed_to_icd10:
+            self.data = self._SNOMED_to_ICD10(snomed_to_icd10_path)
+        if ICD10_3Digit:
+            self.data = self._ICD10_3Digit()
```

Comment:
I think we could add a check in here so it errors usefully if you are passing in snomed_to_icd as a arg but the codelist is already icd.